lighthouse-core/audits/preload-lcp-image.js

```diff
@@ -0,0 +1,167 @@
+      scoreDisplayMode: Audit.SCORING_MODES.NUMERIC,
+    };
+  }
+
+  /**
+   *
+   * @param {LH.Artifacts.NetworkRequest} request
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {Array<LH.Gatherer.Simulation.GraphNode>} initiatorPath
+   * @param {string} lcpImageSource
+   * @return {boolean}
+   */
+  static shouldPreloadRequest(request, mainResource, initiatorPath, lcpImageSource) {
+    const mainResourceDepth = mainResource.redirects ? mainResource.redirects.length : 0;
+
+    // If it's not the request for the LCP image, don't recommend it.
+    if (request.url !== lcpImageSource) return false;
+    // If it's already preloaded, no need to recommend it.
+    if (request.isLinkPreload) return false;
+    // It's not a request loaded over the network, don't recommend it.
+    if (URL.NON_NETWORK_PROTOCOLS.includes(request.protocol)) return false;
+    // It's already discoverable from the main document, don't recommend it.
+    if (initiatorPath.length <= mainResourceDepth) return false;
+    // Finally, return whether or not it belongs to the main frame
+    return request.frameId === mainResource.frameId;
+  }
+
+  /**
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @param {LH.Artifacts.TraceElement|undefined} lcpElement
+   * @param {Array<LH.Artifacts.ImageElement>} imageElements
+   * @return {string|undefined}
+   */
+  static getURLToPreload(mainResource, graph, lcpElement, imageElements) {
+    if (!lcpElement) return undefined;
+
+    const lcpImageElement = imageElements.find(elem => {
+      return elem.devtoolsNodePath === lcpElement.devtoolsNodePath;
+    });
+
+    if (!lcpImageElement) return undefined;
+    let lcpUrl;
+    graph.traverse((node, traversalPath) => {
+      if (node.type !== 'network') return;
+      const record = node.record;
+      const imageSource = lcpImageElement.src;
+      // Don't include the node itself or any CPU nodes in the initiatorPath
+      const path = traversalPath.slice(1).filter(initiator => initiator.type === 'network');
+
+      // eslint-disable-next-line max-len
+      if (!PreloadLCPImageAudit.shouldPreloadRequest(record, mainResource, path, imageSource)) return;
+      lcpUrl = record.url;
+    });
+
+    return lcpUrl;
+  }
+
+  /**
+   * Computes the estimated effect of preloading the LCP image.
+   * @param {string} lcpUrl // The image URL of the LCP
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @param {LH.Gatherer.Simulation.Simulator} simulator
+   * @return {{wastedMs: number, results: Array<{url: string, wastedMs: number}>}}
+   */
+  static computeWasteWithGraph(lcpUrl, graph, simulator) {
+    const simulation = simulator.simulate(graph, {flexibleOrdering: true});
+    // For now, we are simply using the duration of the LCP image request as the wastedMS value
+    const timings = Array.from(simulation.nodeTimings.entries());
```

Comment:
if we had the `lcpNode` object in my previous recommendations we could just do `simulation.nodeTimings.get(lcpNode)` :D